src/amr/multiphysics_integrator.hpp

```diff
@@ -344,6 +344,11 @@ namespace solver
 
             levelInitializer.initialize(hierarchy, levelNumber, oldLevel, model, messenger,
                                         initDataTime, isRegridding);
+
+            if (static_cast<std::size_t>(levelNumber) == model_views_.size())
+                model_views_.push_back(solver.make_view(*level, model));
+            else
+                model_views_[levelNumber] = solver.make_view(*level, model);
```

Comment:
I wasn't super sure of the point of the level descriptor system here so I didn't use it.

That might very well be an issue for future usage with varying solvers between levels, tbc